core/orch_test.go

```diff
@@ -974,8 +977,26 @@ func TestProcessPayment_GivenReceiveTicketError_ReturnsError(t *testing.T) {
 
 	time.Sleep(time.Millisecond * 20)
 	assert := assert.New(t)
-	assert.EqualError(err, "ReceiveTicket error")
+	assert.EqualError(err, errInvalidTicketSignature.Error())
 	recipient.AssertNumberOfCalls(t, "RedeemWinningTicket", 1)
+
+	// Does not loop through tickets if won==false and error is a fatal receive error
+	recipient.On("ReceiveTicket", mock.Anything, mock.Anything, mock.Anything).Return("", false, pm.NewFatalReceiveErr(errors.New("ReceiveTicket error"))).Once()
+	err = orch.ProcessPayment(*defaultPaymentWithTickets(t, senderParams), manifestID)
+	time.Sleep(time.Millisecond * 20)
+	_, ok := err.(*pm.FatalReceiveErr)
+	assert.True(ok)
+	// Still 1 call to RedeemWinningTicket
+	recipient.AssertNumberOfCalls(t, "RedeemWinningTicket", 1)
+
+	// Redeem winning tickets if won==true and not a signature error (but err != nil)
+	recipient.On("ReceiveTicket", mock.Anything, mock.Anything, mock.Anything).Return("", true, errors.New("ReceiveTicket error")).Once()
+	recipient.On("ReceiveTicket", mock.Anything, mock.Anything, mock.Anything).Return("", true, nil).Once()
+	err = orch.ProcessPayment(*defaultPaymentWithTickets(t, senderParams), manifestID)
+	time.Sleep(time.Millisecond * 20)
+	assert.EqualError(err, "ReceiveTicket error")
+	// 3 RedeemWinningTicketCalls (1 + 2)
```

Comment:
"RedeemWinningTicketCalls" -> "RedeemWinningTicket calls"

Reply:
fixed in 6cae502